packages/playwright/src/worker/workerMain.ts

```diff
@@ -384,8 +373,13 @@ export class WorkerMain extends ProcessRunner {
       await testInfo._runWithTimeout({ type: 'test' }, async () => {
         // Now run the test itself.
         const fn = test.fn; // Extract a variable to get a better stack trace ("myTest" vs "TestCase.myTest [as fn]").
-        await fn(testFunctionParams, testInfo);
-        checkForFloatingPromises('the test');
+        try {
+          await fn(testFunctionParams, testInfo);
+        } catch (e) {
+          throw e;
+        } finally {
+          testInfo._checkForFloatingPromises('the test');
```

Comment:
Perhaps only do this on success? A failing test already has some errors to worry about, wdyt? Note I do not feel strongly about this.